src/ts/components/subtitleoverlay.ts

```diff
@@ -236,14 +279,20 @@ class ActiveSubtitleManager {
    * Calculates a unique ID for a subtitle cue, which is needed to associate an ON_CUE_ENTER with its ON_CUE_EXIT
    * event so we can remove the correct subtitle in ON_CUE_EXIT when multiple subtitles are active at the same time.
    * The start time plus the text should make a unique identifier, and in the only case where a collision
-   * can happen, two similar texts will be displayed at a similar time.
+   * can happen, two similar texts will be displayed at a similar time and a similar position (or without position).
    * The start time should always be known, because it is required to schedule the ON_CUE_ENTER event. The end time
    * must not necessarily be known and therefore cannot be used for the ID.
    * @param event
    * @return {string}
    */
   private static calculateId(event: SubtitleCueEvent): string {
-    return event.start + event.text;
+    let id = event.start + '-' + event.text;
+
+    if (event.position) {
+      id += '-' + event.position.row + '-' + event.position.column;
+    }
```

Comment:
Is this required for anything or just in case?

I guess strictly theoretically if two caption happens at the same time one with "test-0-0" as the text and the other with just "test" and a position 0 0 that would still conflict.

Reply:
Without it, it could not display the same text at multiple positions (e.g. "Test" @ 0,0 and "Test" @ 1,0). Both would have the same ID and that screws up cue handling.

Reply:
> if two caption happens at the same time one with "test-0-0" as the text and the other with just "test" and a position 0 0 that would still conflict

That would be a problem anyway because we currently do not support mixing positioned and unpositioned captions. But for ID generation, the positions are only added when there is a position object. It's pre properties inside the objects that get 0 assigned if they are undefined. If the position object is missing at all, it does not get 0-positions assigned.

Reply:
Ok that was more nitpicking anyway as I was surprised since this wasn't mentioned in the customer issue.